### PR TITLE
adds support for QQBar Serialization

### PR DESCRIPTION
**Pull request by antonydellavecchia**
Adds Support for QQBar Serialization

### CODE REVIEW

File: src/Serialization/Fields.jl [hunk 1]
```diff
@@ -532,6 +533,27 @@ function load_object(s::DeserializerState, ::Type{<:EmbeddedNumFieldElem}, paren
...
+    save_object(s, AcbField()(q), :acb )
+  end
+end
+
+function load_object(s::DeserializerState, ::Type{QQBarFieldElem})
+  Qx, x = QQ[:x]
+  CC = AcbField()
+  min_poly = load_object(s, PolyRingElem{QQ}, Qx, :minpoly)
+  approximation = load_object(s, AcbFieldElem, CC, :acb)
+  return filter(x -> overlaps(approximation, CC(x)), roots(QQBarField(), min_poly))[1]
```
Comment (thofma) [inline src/Serialization/Fields.jl]: One should add a check that there is only one thing overlapping (or use `only`). But if this fails, then the original object cannot be recovered. I will push some suggestion to fix this.

Comment (antonydellavecchia) [reply src/Serialization/Fields.jl]: This also means we need to know the precision of CC needs to be just that the filter had a unique element. So we would also want this store this, i.e. add a check unique in the save as well

Comment (thofma) [reply src/Serialization/Fields.jl]: Yes/no. One can ask for as much precision as one wants during the saving. Sou you would look (during the saving) at `approximation AcbField(prec)(q)` and increase the prec until the filter dance returns a unique element. Do you know what I mean? If not, I can push the suggestion.

Comment (antonydellavecchia) [reply src/Serialization/Fields.jl]: Ya I know what you mean. I'll push a change sometime tomorrow

**Comment by benlorenz** [comment]
Can we make sure that this works with polytopes as well, currently I get an error because it tries to save the field itself which is omitted in the code for the PR:
```julia
julia> f = algebraic_closure(QQ)
Field of algebraic numbers

julia> n2 = f(5)^(QQ(4//5))
Root 3.62390 of x^5 - 625

julia> c = cube(f, 3, -1, n2)
Polytope in ambient dimension 3 with QQBarFieldElem type coefficients

julia> save("/tmp/c.mrdi",c)
ERROR: MethodError: no method matching save_object(::Oscar.SerializerState, ::QQBarField)

Closest candidates are:
  save_object(::Oscar.SerializerState, ::Any, ::Symbol)
   @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:137
  save_object(::Oscar.SerializerState, ::FpFieldElem)
   @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/Fields.jl:108
  save_object(::Oscar.SerializerState, ::FinGenAbGroup)
   @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/Groups.jl:238
  ...

Stacktrace:
  [1] save_object(s::Oscar.SerializerState, x::QQBarField, key::Symbol)
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:139
  [2] (::Oscar.var"#6044#6045"{Oscar.SerializerState, Dict{String, Any}})()
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/containers.jl:344
  [3] #5924
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:97 [inlined]
  [4] serialize_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:74 [inlined]
  [5] save_data_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:95 [inlined]
  [6] save_data_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:94 [inlined]
  [7] save_object
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/containers.jl:342 [inlined]
  [8] save_object(s::Oscar.SerializerState, x::Dict{String, Any}, key::Symbol)
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:139
  [9] save_typed_object(s::Oscar.SerializerState, x::Dict{String, Any})
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:162
 [10] #6060
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/PolyhedralGeometry.jl:52 [inlined]
 [11] #5924
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:97 [inlined]
 [12] serialize_dict(f::Oscar.var"#5924#5925"{Oscar.var"#6060#6061"{…}, Oscar.SerializerState}, s::Oscar.SerializerState)
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:74
 [13] save_data_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:95 [inlined]
 [14] save_data_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:94 [inlined]
 [15] save_object
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/PolyhedralGeometry.jl:51 [inlined]
 [16] save_object
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:139 [inlined]
 [17] save_typed_object(s::Oscar.SerializerState, x::Polyhedron{QQBarFieldElem})
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:162
 [18] (::Oscar.var"#6382#6385"{Polyhedron{…}, Nothing, DataType, Polyhedron{…}, Oscar.SerializerState})()
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:427
 [19] #5924
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:97 [inlined]
 [20] serialize_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:74 [inlined]
 [21] save_data_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:95 [inlined]
 [22] save_data_dict
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/serializers.jl:94 [inlined]
 [23] #save#6381
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:423 [inlined]
 [24] save
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:420 [inlined]
 [25] #6389
    @ ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:465 [inlined]
 [26] open(::Oscar.var"#6389#6390"{Nothing, Polyhedron{QQBarFieldElem}}, ::String, ::Vararg{String}; kwargs::@Kwargs{})
    @ Base ./io.jl:396
 [27] open
    @ ./io.jl:393 [inlined]
 [28] save(filename::String, obj::Polyhedron{QQBarFieldElem}; metadata::Nothing)
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:464
 [29] save(filename::String, obj::Polyhedron{QQBarFieldElem})
    @ Oscar ~/software/polymake/julia/Oscar.jl/src/Serialization/main.jl:460
 [30] top-level scope
    @ REPL[29]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

**Comment by thofma** [comment]
I just added
```
function save_object(s::SerializerState, q::QQBarField)
end

function load_object(s::DeserializerState, ::Type{QQBarField})
  return QQBarField()
end
```
but it seems that the serialization framework does not like this :)

The example of @benlorenz is failing with
```
julia> save("/tmp/c.mrdi",c)
ERROR: ArgumentError: Key :_coeff is being overridden by :_description before write.
Stacktrace:
  [1] macro expansion
    @ ~/bla/src/Assertions.jl:561 [inlined]
  [2] set_key
    @ ~/bla/src/Serialization/serializers.jl:85 [inlined]
  [3] save_object(s::Oscar.SerializerState, x::String, key::Symbol)
    @ Oscar ~/bla/src/Serialization/main.jl:138
```
It works if one does
```
function save_object(s::SerializerState, q::QQBarField)
  save_object(s, 1)
end

function load_object(s::DeserializerState, ::Type{QQBarField})
  return QQBarField()
end
```
but I let @antonydellavecchia decide how he wants to proceed.

**Review by benlorenz — APPROVED** [review]

**Pull request merged**